src/backend/mps/compute.rs

```diff
@@ -18,23 +23,408 @@ impl MpsCompute {
+            height: 1,
+            depth: 1,
+        };
+        let thread_group_size = MTLSize {
+            width: num_threads,
+            height: 1,
+            depth: 1,
+        };
+
+        compute_encoder.dispatch_thread_groups(thread_group_count, thread_group_size);
+        compute_encoder.end_encoding();
+
+        command_buffer.commit();
+        command_buffer.wait_until_completed();
+
+        Ok(result_buffer)
+    }
+
+    pub fn sub(&self, a: &Buffer, b: &Buffer, size: usize) -> Result<Buffer, MpsError> {
+        let result_buffer = self.device.device().new_buffer(
+            (size * std::mem::size_of::<f32>()) as u64,
+            MTLResourceOptions::StorageModeShared,
+        );
+
+        // Create and compile the addition kernel
+        let library = self
+            .device
+            .device()
+            .new_library_with_source(
+                include_str!("../../../shaders/metal/binary_ops.metal"),
+                &metal::CompileOptions::new(),
+            )
+            .map_err(|_| MpsError::ShaderCompilationError)?;
+
+        let kernel = library
+            .get_function("vector_sub", None)
+            .map_err(|_| MpsError::ShaderCompilationError)?;
+
+        let pipeline = self
+            .device
+            .device()
+            .new_compute_pipeline_state_with_function(&kernel)
+            .map_err(|_| MpsError::ShaderCompilationError)?;
+
+        let command_buffer = self.command_queue.new_command_buffer();
+        let compute_encoder = command_buffer.new_compute_command_encoder();
+
+        compute_encoder.set_compute_pipeline_state(&pipeline);
+        compute_encoder.set_buffer(0, Some(a), 0);
+        compute_encoder.set_buffer(1, Some(b), 0);
+        compute_encoder.set_buffer(2, Some(&result_buffer), 0);
+
+        let num_threads = pipeline.thread_execution_width();
+
+        // Create thread groups
+        let thread_group_count = MTLSize {
+            width: ((size as NSUInteger + num_threads) / num_threads),
+            height: 1,
+            depth: 1,
+        };
+        let thread_group_size = MTLSize {
+            width: num_threads,
+            height: 1,
+            depth: 1,
+        };
+
+        compute_encoder.dispatch_thread_groups(thread_group_count, thread_group_size);
+        compute_encoder.end_encoding();
+
+        command_buffer.commit();
+        command_buffer.wait_until_completed();
+
+        Ok(result_buffer)
+    }
+
+    pub fn multiply(&self, a: &Buffer, b: &Buffer, size: usize) -> Result<Buffer, MpsError> {
+        let result_buffer = self.device.device().new_buffer(
+            (size * std::mem::size_of::<f32>()) as u64,
+            MTLResourceOptions::StorageModeShared,
+        );
+
+        let library = self
+            .device
+            .device()
+            .new_library_with_source(
+                include_str!("../../../shaders/metal/binary_ops.metal"),
```

Comment:
Is there a way to transfer this section to `build.rs`?

Reply:
i don't know about that section can be inside of `build.rs`.

i think you want build metal files(to metallib).
but it could be same to using MSL (Metal Shading Language, .metal extension file), except it didn't compile on runtime
See the [metal-rs sample code with metallib](https://github.com/gfx-rs/metal-rs/blob/master/examples/mps/main.rs)

Alternative way is that code could be inside of compute struct new method to reusing library variable on compute functions(ex. matmul).
And to better performance, should be .metal file compiles on compile time not in the runtime (using metallib instead metal files). 
As you expect, in the a build.rs.